src/browser-frontend/views/browser/window.jsx

```diff
@@ -10,17 +10,48 @@ CONDITIONS OF ANY KIND, either express or implied. See the License for the
 specific language governing permissions and limitations under the License.
 */
 
-import React, { PureComponent } from 'react';
+import React, { PureComponent, PropTypes } from 'react';
 import CSSModules from 'react-css-modules';
+import { connect } from 'react-redux';
+import Mousetrap from 'mousetrap';
+
+import { client } from '../../global';
+import PagesModelActions from '../../actions/pages-model-actions';
+import * as SharedPropTypes from '../../model/shared-prop-types';
+import SharedActions from '../../../shared/actions/shared-actions';
+import * as UIPagesSelectors from '../../selectors/ui-pages-selectors';
+
 
 import Styles from './window.css';
 import Chrome from './chrome';
 import Content from './content';
 
+@connect(state => ({
+  client,
+  selectedPageId: UIPagesSelectors.getSelectedPageId(state),
```

Comment:
I'm noticing this now, but `selectedPageId` isn't used in rendering. Having props in a component which aren't used for rendering is discouraged. I. this case, for example, the Window component will re-render every time the `selectedPageId` prop changes (so... on every tab change). This probably isn't something noticeably affecting performance right now, but these kinds of things add up quickly.

Instead, use a thunk as described above (`this.props.dispatch((dispatch, getState) => ...`) to get any relevant state. I'll add a test in the future checking against this antipattern.